src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All side lengths must be greater than zero');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
+    }
```

Comment:
The constructor checks for the validity of the triangle sides using the parameters `a`, `b`, and `c`. However, these parameters should be accessed using `this.a`, `this.b`, and `this.c` since they are class properties. This will ensure that the checks are performed on the instance variables rather than the constructor parameters.